### PR TITLE
[SMALLFIX] Added test for FileWorkerMasterSyncExecutor

### PR DESCRIPTION
**Pull request by yupeng9**

### CODE REVIEW

File: servers/src/test/java/tachyon/worker/file/FileWorkerMasterSyncExecutorTest.java [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+  @Before
+  public void before() {
+    mFileDataManager = Mockito.mock(FileDataManager.class);
+    mFileSystemMasterClient = Mockito.mock(FileSystemMasterClient.class);
+    mFileWorkerMasterSyncExecutor =
+        new FileWorkerMasterSyncExecutor(mFileDataManager, mFileSystemMasterClient);
+  }
+
+  /**
+   * Ensure {@link FileDataManager.#clearPersistedFiles(java.util.List)} is only called when the
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/file/FileWorkerMasterSyncExecutorTest.java]: Ensures

File: servers/src/test/java/tachyon/worker/file/FileWorkerMasterSyncExecutorTest.java [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+   * heartbeat of {@link FileSystemMasterClient} does not fail.
+   */
+  @Test
+  public void heartbeatFailureTest() throws Exception {
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/file/FileWorkerMasterSyncExecutorTest.java]: Can we break this two unit tests, one is heartbeatFailureTest one is heatbeatSuccessTest? Ideally, each unit test should be as simple and as as basic as possible IMO.

**Comment by apc999** [comment]
left two comments. thanks for improving the test coverage.

File: servers/src/test/java/tachyon/worker/file/FileWorkerMasterSyncExecutorTest.java [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+    List<Long> persistedFiles = Lists.newArrayList(1L);
+    Mockito.when(mFileDataManager.getPersistedFiles()).thenReturn(persistedFiles);
+    // first time fails, second time passes
+    Mockito.when(mFileSystemMasterClient.heartbeat(WorkerIdRegistry.getWorkerId(), persistedFiles))
+        .thenThrow(new IOException("failure"));
+    mFileWorkerMasterSyncExecutor.heartbeat();
+    Mockito.verify(mFileDataManager, Mockito.never()).clearPersistedFiles(persistedFiles);
+  }
+
+  @Test
```
Comment (apc999) [inline servers/src/test/java/tachyon/worker/file/FileWorkerMasterSyncExecutorTest.java]: can you add a javadoc on this?
thanks

**Comment by apc999** [comment]
The PR looks good to me. I left one comment on the javadoc part

**Comment by yupeng9** [comment]
@apc999 doc added, PTAL

**Comment by apc999** [comment]
+1 LGTM

**Pull request merged**